http/src/main/resources/init-resources/gce-init.sh

```diff
@@ -188,116 +192,73 @@ USER_SCRIPT_OUTPUT_URI=$(userScriptOutputUri)
 JUPYTER_NOTEBOOK_FRONTEND_CONFIG_URI=$(jupyterNotebookFrontendConfigUri)
 CUSTOM_ENV_VARS_CONFIG_URI=$(customEnvVarsConfigUri)
 RSTUDIO_LICENSE_FILE=$(rstudioLicenseFile)
+GPU_ENABLED=$(gpuEnabled)
 
-mkdir -p /work
-mkdir -p /certs
+CERT_DIRECTORY='/var/certs'
+DOCKER_COMPOSE_FILES_DIRECTORY='/var/docker-compose-files'
+WORK_DIRECTORY='/mnt/disks/work'
+GSUTIL_CMD='docker run --rm -v /var:/var gcr.io/google-containers/toolbox:20200603-00 gsutil'
+GCLOUD_CMD='docker run --rm -v /var:/var gcr.io/google-containers/toolbox:20200603-00 gcloud'
+DOCKER_COMPOSE='docker run --rm -v /var/run/docker.sock:/var/run/docker.sock -v /var:/var docker/compose:1.29.1'
+
+mkdir -p ${WORK_DIRECTORY}
+mkdir -p ${CERT_DIRECTORY}
+mkdir -p ${DOCKER_COMPOSE_FILES_DIRECTORY}
 
 log 'Formatting and mounting persistent disk...'
 
 # Format and mount persistent disk
-export DISK_DEVICE_ID=$(lsblk -o name,serial | grep 'user-disk' | awk '{print $1}')
-# Only format disk is it hasn't already been formatted
+# Fix this to `sdb`. We've never seen a device name that's not `sdb`,
+# Altho you some images, this cmd $(lsblk -o name,serial | grep 'user-disk' | awk '{print $1}')
+# can be used to find device name, this doesn't work for COS images
+export DISK_DEVICE_ID='sdb'
+## Only format disk is it hasn't already been formatted
 if [ "$IS_GCE_FORMATTED" == "false" ] ; then
   mkfs.ext4 -m 0 -E lazy_itable_init=0,lazy_journal_init=0,discard /dev/${DISK_DEVICE_ID}
 fi
-mount -o discard,defaults /dev/${DISK_DEVICE_ID} /work
-# Ensure persistent disk re-mounts if runtime stops and restarts
-cp /etc/fstab /etc/fstab.backup
-echo UUID=`blkid -s UUID -o value /dev/${DISK_DEVICE_ID}` /work ext4 discard,defaults,nofail 0 2 | tee -a /etc/fstab
-chmod a+rwx /work
 
 # done persistent disk setup
 STEP_TIMINGS+=($(date +%s))
 
+if [ "${GPU_ENABLED}" == "true" ] ; then
+  log 'Installing GPU driver...'
+  cos-extensions install gpu
+  mount --bind /var/lib/nvidia /var/lib/nvidia
+  mount -o remount,exec /var/lib/nvidia
+
+  $GSUTIL_CMD cp ${GPU_DOCKER_COMPOSE} ${DOCKER_COMPOSE_FILES_DIRECTORY}
+fi
+
 log 'Copying secrets from GCS...'
 
 # Add the certificates from the bucket to the VM. They are used by the docker-compose file
-gsutil cp ${SERVER_CRT} /certs
-gsutil cp ${SERVER_KEY} /certs
-gsutil cp ${ROOT_CA} /certs
-gsutil cp ${PROXY_SITE_CONF} /etc
-gsutil cp ${JUPYTER_DOCKER_COMPOSE_GCE} /etc
-gsutil cp ${RSTUDIO_DOCKER_COMPOSE} /etc
-gsutil cp ${PROXY_DOCKER_COMPOSE} /etc
-gsutil cp ${WELDER_DOCKER_COMPOSE} /etc
-gsutil cp ${CRYPTO_DETECTOR_DOCKER_COMPOSE} /etc
-
-echo "" > /etc/google_application_credentials.env
+$GSUTIL_CMD cp ${SERVER_CRT} ${CERT_DIRECTORY}
```

Comment:
(Minor) since all these files are in the same dir in the bucket, I wonder if we could do something like:
```
gsutil cp gs://init-bucket/* /var/
```

That way we wouldn't have to spin up the `gsutil` container multiple times, which might be faster.

I don't feel too strongly to do it in this PR, just an idea.